core/src/ast_builder/table_factor.rs

```diff
@@ -19,6 +19,7 @@ pub struct TableFactorNode<'a> {
     pub table_name: String,
     pub table_type: TableType<'a>,
     pub table_alias: Option<String>,
+    pub index: Option<IndexItem>,
```

Comment:
When we use ast builder, all the ast building process can be made without failure which means that all a kind of `build` can be done in lazy.
let's consider to have `IndexItemNode`.